src/libs/actions/Report.js

```diff
@@ -44,6 +44,16 @@ Onyx.connect({
     callback: val => myPersonalDetails = val,
 });
 
+const allReports = {};
+Onyx.connect({
+    key: ONYXKEYS.COLLECTION.REPORT,
+    callback: (val) => {
+        if (val && val.reportID) {
+            allReports[val.reportID] = val;
+        }
+    },
+});
```

Comment:
NAB, I think we have avoided storing all the report data because of an overhead to storing all reports in memory. Not sure if it matters or not, but we are only accessing the report to get the `participants` so it might make more sense to store only what we need instead of everything.

Reply:
I can take a look at this once separately. 

Presumable we would switch to a reportParticipants object, with reportID as the key and participants as the value?

Reply:
Yep that's what I was thinking!

Reply:
Resolving, will implement this change in the [IOU Details PR](https://github.com/Expensify/Expensify/issues/158811)